src/component.js

```diff
@@ -332,13 +336,19 @@ var QuillComponent = React.createClass({
 	},
 
 	onEditorChangeSelection: function(range, source, editor) {
-		var s = this.getEditorSelection() || {};
-		var r = range || {};
-		if (r.length !== s.length || r.index !== s.index) {
-			this.setState({ selection: range });
-			if (this.props.onChangeSelection) {
-				this.props.onChangeSelection(range, source, editor);
-			}
+		var currentSelection = this.getEditorSelection();
+		var nextSelection = range;
+		if (isEqual(nextSelection, currentSelection)) {
+			return;
+		}
+		this.setState({ selection: range });
+		if (this.props.onChangeSelection) {
+			this.props.onChangeSelection(range, source, editor);
+		}
+		if (this.props.onFocus && !currentSelection && nextSelection) {
```

Comment:
I improved the readability of this code using your suggestion.

I also think it makes sense to have empty placeholders as default handlers, however it's out of scope of this one. But I do agree.